packages/contracts-bedrock/src/deploy-config.ts

```diff
@@ -0,0 +1,339 @@
+  gasPriceOracleOverhead: number
+  gasPriceOracleScalar: number
+  gasPriceOracleDecimals: number
+  numDeployConfirmations: number
+}
+
+/**
+ * Full deployment configuration.
+ */
+export type DeployConfig = RequiredDeployConfig &
+  Partial<OptionalL1DeployConfig> &
+  Partial<OptionalL2DeployConfig>
+
+/**
+ * Deployment configuration specification for the hardhat plugin.
+ */
+export const deployConfigSpec: {
+  [K in keyof DeployConfig]: {
+    type: 'string' | 'number' | 'boolean' | 'address'
+    default?: any
+  }
+} = {
+  finalSystemOwner: {
+    type: 'address',
+    default: ethers.constants.AddressZero,
+  },
+  controller: {
+    type: 'address',
+    default: ethers.constants.AddressZero,
+  },
+  l1StartingBlockTag: {
+    type: 'string',
+  },
+  l1ChainID: {
+    type: 'number',
+  },
+  l2ChainID: {
+    type: 'number',
+  },
+  l2BlockTime: {
+    type: 'number',
+  },
+  maxSequencerDrift: {
+    type: 'number',
+  },
+  sequencerWindowSize: {
+    type: 'number',
+  },
+  channelTimeout: {
+    type: 'number',
+  },
+  p2pSequencerAddress: {
+    type: 'address',
+  },
+  batchInboxAddress: {
+    type: 'address',
+  },
+  batchSenderAddress: {
+    type: 'address',
+  },
+  l2OutputOracleSubmissionInterval: {
+    type: 'number',
+  },
+  l2OutputOracleStartingBlockNumber: {
+    type: 'number',
+    default: 0,
+  },
+  l2OutputOracleStartingTimestamp: {
+    type: 'number',
+  },
+  l2OutputOracleProposer: {
+    type: 'address',
+  },
+  l2OutputOracleChallenger: {
+    type: 'address',
+  },
+  finalizationPeriodSeconds: {
+    type: 'number',
+    default: 2,
+  },
+  systemConfigOwner: {
+    type: 'address',
+  },
+  l1BlockTime: {
+    type: 'number',
+    default: 15,
+  },
+  l1GenesisBlockNonce: {
+    type: 'string', // uint64
+    default: '0x0',
+  },
+  cliqueSignerAddress: {
+    type: 'address',
+    default: ethers.constants.AddressZero,
+  },
+  l1GenesisBlockGasLimit: {
+    type: 'string',
+    default: ethers.BigNumber.from(15_000_000).toHexString(),
+  },
+  l1GenesisBlockDifficulty: {
+    type: 'string', // uint256
+    default: '0x1',
+  },
+  l1GenesisBlockMixHash: {
+    type: 'string', // bytes32
+    default: ethers.constants.HashZero,
+  },
+  l1GenesisBlockCoinbase: {
+    type: 'address',
+    default: ethers.constants.AddressZero,
+  },
+  l1GenesisBlockNumber: {
+    type: 'string', // uint64
+    default: '0x0',
+  },
+  l1GenesisBlockGasUsed: {
+    type: 'string', // uint64
+    default: '0x0',
+  },
+  l1GenesisBlockParentHash: {
+    type: 'string', // bytes32
+    default: ethers.constants.HashZero,
+  },
+  l1GenesisBlockBaseFeePerGas: {
+    type: 'string', // uint256
+    default: ethers.BigNumber.from(1000_000_000).toHexString(), // 1 gwei
+  },
+  l2GenesisBlockNonce: {
+    type: 'string', // uint64
+    default: '0x0',
+  },
+  l2GenesisBlockExtraData: {
+    type: 'string', // important: in the case of L2, which uses post-Merge Ethereum rules, this must be <= 32 bytes.
```

Comment:
If there was a `bytes32` type that would be helpful here but the comment is good enough

Reply:
I can add that later